kernel/kernel/fs/pipe.cpp

```diff
@@ -1040,4 +1149,28 @@ TEST(pipe, broken_pipe)
     EXPECT_EQ(p->get_unread_len(), 0U);
 }
 
+TEST(pipe, readv_works)
+{
+    auto_addr_limit l_{VM_KERNEL_ADDR_LIMIT};
+    auto p = make_refc<pipe>();
+    p->reader_count = 1;
+    p->writer_count = 1;
+
+    char teststr[] = { '1', '1', '1', '1' };
+    p->write(0, 2, teststr);
+
+    iovec iov[2];
+    iov[0].iov_base = teststr;
+    iov[0].iov_len = 2;
+    iov[1].iov_base = teststr + 2;
+    iov[1].iov_len = 2;
+
+    cul::slice<iovec> sl(iov, 2);
+    iovec_iter iter(sl, 4, IOVEC_KERNEL);
+
+    ssize_t total_read = p->read_iter(&iter, O_NONBLOCK);
+
+    EXPECT_EQ(total_read, 2);
+}
+
```

Comment:
Test LGTM

Reply:
Wdym?

Reply:
Test Looks Good To Me ;)

Reply:
I still don't get what should I do here

Reply:
argh you should do nothing

stupid stupid github pull requests fuck this